flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/BufferManager.java

```diff
@@ -349,7 +373,7 @@ int unsynchronizedGetFloatingBuffersAvailable() {
 		 */
 		int addExclusiveBuffer(Buffer buffer, int numRequiredBuffers) {
 			exclusiveBuffers.add(buffer);
-			if (getAvailableBufferSize() > numRequiredBuffers) {
+			if (numRequiredBuffers == 0) {
```

Comment:
It should not be changed here. If the `numRequiredBuffers` is 0, `getAvailableBufferSize()` must be more than it. If `numRequiredBuffers` is `1` or something else, as long as the `getAvailableBufferSize()` is more than it, we also need to release a floating buffer.